x-pack/plugins/maps/public/routing/routes/maps_app/load_map_and_render.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { MapsAppView } from '.';
+import { getMapsSavedObjectLoader } from '../../bootstrap/services/gis_map_saved_object_loader';
+import { getToasts } from '../../../kibana_services';
+import { i18n } from '@kbn/i18n';
+import { Redirect } from 'react-router-dom';
+
+export const LoadMapAndRender = class extends React.Component {
+  state = {
+    savedMap: null,
+    failedToLoad: false,
+  };
+
+  componentDidMount() {
+    this.loadSavedMap();
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async loadSavedMap() {
+    const { savedMapId } = this.props.match.params;
+    try {
+      const savedMap = await getMapsSavedObjectLoader().get(savedMapId);
+      if (this._isMounted) {
+        this.setState({ savedMap });
+      }
+    } catch (err) {
+      if (this._isMounted) {
+        this.setState({ failedToLoad: true });
+      }
+      getToasts().addWarning({
```

Comment:
Can you move toast under isMounted check. If user navigates away then no point showing an error message.